cmd/soroban-rpc/internal/db/transaction.go

```diff
@@ -142,32 +142,45 @@ func (txn *transactionHandler) GetLedgerRange(ctx context.Context) (ledgerbucket
 	// and max from the ledger table in a single query and get around sqlite's
 	// limitations with parentheses (see https://stackoverflow.com/a/22609948).
 	//
-	newestQ := sq.
-		Select("m1.meta").
+	// Queries to get the minimum and maximum ledger sequence from the transactions table
+	minLedgerSeqQ := sq.
+		Select("m1.ledger_sequence").
 		FromSelect(
 			sq.
-				Select("meta").
-				From(ledgerCloseMetaTableName).
-				OrderBy("sequence ASC").
+				Select("ledger_sequence").
+				From(transactionTableName).
+				OrderBy("ledger_sequence ASC").
 				Limit(1),
 			"m1",
 		)
-	sql, args, err := sq.
-		Select("m2.meta").
+	maxLedgerSeqQ, args, err := sq.
+		Select("m2.ledger_sequence").
 		FromSelect(
 			sq.
-				Select("meta").
-				From(ledgerCloseMetaTableName).
-				OrderBy("sequence DESC").
+				Select("ledger_sequence").
+				From(transactionTableName).
+				OrderBy("ledger_sequence DESC").
 				Limit(1),
 			"m2",
 		).ToSql()
 	if err != nil {
-		return ledgerRange, errors.Wrap(err, "couldn't build ledger range query")
+		return ledgerRange, errors.Wrap(err, "couldn't build max ledger sequence query")
 	}
 
+	// Combine the min and max ledger sequence queries using UNION ALL
+	txnMinMaxLedgersQ, _, err := minLedgerSeqQ.Suffix("UNION ALL "+maxLedgerSeqQ, args...).ToSql()
+	if err != nil {
+		return ledgerRange, errors.Wrap(err, "couldn't build max ledger sequence query")
+	}
+
+	// Final query to join ledger_close_meta table and the sequence numbers we got from txnMinMaxLedgersQ
+	finalSql := sq.
+		Select("lcm.meta").
+		From(fmt.Sprintf("%s as lcm", ledgerCloseMetaTableName)).
+		JoinClause(fmt.Sprintf("JOIN (%s) as seqs ON lcm.sequence == seqs.ledger_sequence", txnMinMaxLedgersQ))
+
 	var lcms []xdr.LedgerCloseMeta
-	if err = txn.db.Select(ctx, &lcms, newestQ.Suffix("UNION ALL "+sql, args...)); err != nil {
+	if err = txn.db.Select(ctx, &lcms, finalSql); err != nil {
```

Comment:
This seems overly complicated. I think you can use the [min/max functions](https://www.sqlite.org/lang_aggfunc.html) to drastically simplify this.

Also, please benchmark this and make sure it takes a reasonable time.

Reply:
Unfortunately you can't. Well, you can, but you cannot combine `min` and `max` it the same query because we also need the respective `meta` from the ledgers table to get the ledger close time alongside the sequence number. So you have to have separate `ORDER BY LIMIT 1`s. You can also use a subquery. @tamirms and I discussed this here: https://github.com/stellar/soroban-rpc/pull/141#discussion_r1578433440.

Also there are benchmarks for getTransactionByHash which tests this by extension (see https://github.com/stellar/soroban-rpc/pull/141#issuecomment-2088676124).

Reply:
I am pretty sure you can use max/min as a subquery and then a join

Reply:
I would like an explicit yes for this, and some numbers in the PR